app/src/main/java/de/westnordost/streetcomplete/screens/about/LogsFiltersDialog.kt

```diff
@@ -0,0 +1,163 @@
+    private val binding = DialogLogsFiltersBinding.inflate(LayoutInflater.from(context))
+
+    init {
+        setView(binding.root)
+
+        setButton(BUTTON_POSITIVE, context.getString(R.string.action_filter)) { _, _ ->
+            onApplyButtonClick(filters)
+            dismiss()
+        }
+        setButton(BUTTON_NEGATIVE, context.getString(R.string.action_reset)) { _, _ ->
+            onApplyButtonClick(LogsFilters())
+            cancel()
+        }
+
+        val allLevels = LogLevel.values().toList()
+        binding.levelList.adapter = LevelsFilterAdapter(
+            allLevels,
+            filters.levels
+        ) {
+            when (filters.levels.contains(it)) {
+                true -> filters.levels.remove(it)
+                false -> filters.levels.add(it)
+            }
+        }
+
+        binding.messageContainsEditText.setText(filters.messageContains)
+        updateMessageContainsClearButton()
+        binding.messageContainsEditText.doAfterTextChanged {
+            filters.messageContains = binding.messageContainsEditText.nonBlankTextOrNull
+            updateMessageContainsClearButton()
+        }
+        binding.messageContainsClearButton.setOnClickListener {
+            filters.messageContains = null
+            binding.messageContainsEditText.setText(null)
+            updateMessageContainsClearButton()
+        }
+
+        updateNewerThanInput()
+        binding.newerThanTextDate.setOnClickListener {
+            pickLocalDateTime(filters.timestampNewerThan ?: LocalDateTime.now()) {
+                filters.timestampNewerThan = it
+                updateNewerThanInput()
+            }
+        }
+        binding.newerThanClearButton.setOnClickListener {
+            filters.timestampNewerThan = null
+            updateNewerThanInput()
+        }
+
+        updateOlderThanInput()
+        binding.olderThanTextDate.setOnClickListener {
+            pickLocalDateTime(filters.timestampOlderThan ?: LocalDateTime.now()) {
+                filters.timestampOlderThan = it
+                updateOlderThanInput()
+            }
+        }
+        binding.olderThanClearButton.setOnClickListener {
+            filters.timestampOlderThan = null
+            updateOlderThanInput()
+        }
+    }
+    private fun updateMessageContainsClearButton() {
+        binding.messageContainsClearButton.visibility = if (filters.messageContains == null) View.GONE else View.VISIBLE
+    }
+
+    private fun updateNewerThanInput() {
+        binding.newerThanClearButton.visibility = if (filters.timestampNewerThan == null) View.GONE else View.VISIBLE
+        binding.newerThanTextDate.setText(filters.timestampNewerThan?.toString() ?: "")
+    }
+
+    private fun updateOlderThanInput() {
+        binding.olderThanClearButton.visibility = if (filters.timestampOlderThan == null) View.GONE else View.VISIBLE
+        binding.olderThanTextDate.setText(filters.timestampOlderThan?.toString() ?: "")
```

Comment:
Timestamps should be properly formatted according to the user's locale. Err... not sure if `kotlinx.time` has something for that, after all, `Locale` is a Java-only class. For Java's old time-date classes, there was `SimpleDateFormat`, but maybe there is (at least) something for the new data-time classes?

Reply:
Unfortunately i18n seems to be out of scope for [`kotlinx-datetime`](https://github.com/Kotlin/kotlinx-datetime#design-overview):
> The library is based on the ISO 8601 international standard, other ways to represent dates and times are out of its scope. Internationalization (such as locale-specific month and day names) is out the scope, too.

I've used `java.time.format.DateTimeFormatter` for now.

Just to be clear, I've only changed formatting of date time filter inputs, but left time labels from the logs list as-is (ISO 8601 formatted). Regardless of user's locale, `HH:mm:ss.SSS` format seems to be the only one that fits limited space and is precise up to milliseconds, don't you think?

Reply:
😥

Looks like we need to look for something for this in the future, then. There is certainly the need for that.

> Regardless of user's locale, HH:mm:ss.SSS format seems to be the only one that fits limited space and is precise up to milliseconds, don't you think?

It should be well understandable at least. And I would think that times are specified like this everywhere anyway (except maybe the ms part).